src/libutil/fs-sink.hh

```diff
@@ -0,0 +1,49 @@
+#pragma once
+///@file
+
+#include "types.hh"
+#include "serialise.hh"
+
+namespace nix {
+
+/**
+ * \todo Fix this API, it sucks.
+ */
+struct ParseSink
```

Comment:
`FileSystemVisitor`?

Reply:
This second PR I guess I hadn't rebased. That stuff is no already there on master, but we could still rename it.